src/live_restore/live_restore_fs.c

```diff
@@ -1920,7 +1946,19 @@ __wt_os_live_restore_fs(
     if (!__wt_ispo2((uint32_t)lr_fs->read_size))
         WT_ERR_MSG(session, EINVAL, "the live restore read size must be a power of two");
 
-    /* Update the callers pointer. */
+    WT_ERR(__wt_spin_init(session, &lr_fs->state_lock, "live restore state lock"));
+
+    /*
+     * To initialize the live restore file system we need to read its state from the turtle file,
```

Comment:
This feels a little clunky, but I don't see a better workaround for this short of re-implementing the turtle_read logic.

Reply:
I am okay with this, it doesn't seem overly clunky.

Reply:
Seems okay to me too. It is well documented/commented and makes sense to the reader.